eng/_core/cmd/build/build.go

```diff
@@ -271,18 +288,26 @@ func getMaxAttemptsOrExit(varName string, defaultValue int) int {
 }
 
 func getEnvIntOrDefault(varName string, defaultValue int) (int, error) {
-	a, ok := os.LookupEnv(varName)
-	if !ok {
-		return defaultValue, nil
-	}
-	if a == "" {
-		return 0, fmt.Errorf(
-			"env var %q is set to empty string, which is not an int. To use the default value %v, unset the env var",
-			varName, defaultValue)
+	a, err := getEnvOrDefault(varName, strconv.Itoa(defaultValue))
+	if err != nil {
+		return 0, err
 	}
 	i, err := strconv.Atoi(a)
 	if err != nil {
 		return 0, fmt.Errorf("env var %q is not an int: %w", varName, err)
 	}
 	return i, nil
 }
+
+func getEnvOrDefault(varName, defaultValue string) (string, error) {
+	v, ok := os.LookupEnv(varName)
+	if !ok {
+		return defaultValue, nil
+	}
+	if v == "" {
```

Comment:
Not sure this is a valid code path. Environment variables in Windows can't be empty strings. Using an empty string as an environment variable value is the equivalent of deleting it. Is this for non-Windows code paths?

Reply:
Yep, this will only happen on non-Windows. It's just a sanity check: https://github.com/microsoft/go/pull/475#discussion_r826666904

> rejecting empty string _could_ help if you're trying to set it like `GO_MAKE_MAX_RETRY_ATTEMPTS=$(fun)`, and for some reason `fun` fails rather than sending anything to stdout.